wp-content/themes/core/components/header/subheader_archive/Subheader_Archive_Controller.php

```diff
@@ -0,0 +1,159 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Project\Templates\Components\header\subheader_archive;
+
+use Tribe\Libs\Utils\Markup_Utils;
+use Tribe\Project\Templates\Components\Abstract_Controller;
+use Tribe\Project\Templates\Components\image\Image_Controller;
+use Tribe\Project\Templates\Components\text\Text_Controller;
+use Tribe\Project\Theme\Config\Image_Sizes;
+
+class Subheader_Archive_Controller extends Abstract_Controller {
+
+	public const CLASSES           = 'classes';
+	public const ATTRS             = 'attrs';
+	public const CONTAINER_CLASSES = 'container_classes';
+	public const MEDIA_CLASSES     = 'media_classes';
+	public const CONTENT_CLASSES   = 'content_classes';
+
+	private array $classes;
+	private array $attrs;
+	private array $container_classes;
+	private array $media_classes;
+	private array $content_classes;
+	private string $title;
+	private string $description;
+	private int $hero_image_id;
+
+	public function __construct( array $args = [] ) {
+		$args = $this->parse_args( $args );
+
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->container_classes = (array) $args[ self::CONTAINER_CLASSES ];
+		$this->media_classes     = (array) $args[ self::MEDIA_CLASSES ];
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->title             = '';
+		$this->description       = '';
+		$this->map_acf_fields();
+	}
+
+	protected function defaults(): array {
+		return [
+			self::CLASSES           => [ 'c-subheader' ],
+			self::CONTAINER_CLASSES => [ 'l-container' ], 
```

Comment:
No need to duplicate the require classes as defaults, too.

Reply:
✅